R/subtotals-and-headings.R

```diff
@@ -214,6 +206,22 @@ setMethod("subtotals<-", c("CrunchVariable", "ANY"), function (x, value) {
         }, value)))) {
         halt("value must be a list of Subtotals, Headings, or both.")
     }
+    # If position is relative we default to setting after to be the last category
+    # in the Subtotal category.
+    value <- lapply(value, function(a){
+        if (is.null(a$after) && a$position == "relative") {
+            if (is.numeric(a$categories)) {
+                var_cats <- ids(categories(x))
+            } else {
+                var_cats <- names(categories(x))
+            }
+            ordered_cats <- a$categories[order(match( a$categories, var_cats))]
```

Comment:
What happens if the category specified in the subtotal is not in the categories in `var_cats`? That shouldn't be a common condition, but I could imagine it happening if someone is applying the same `Subtotal` sets to different variables that have mostly the same, but some different categories.

Also, what happens if the last category happens to be a missing category? I don't see a test for that case, but it would be a good one to cover.

Reply:
Actually, it looks like having a missing category in the subtotal _is_ covered by the test (line 129), but having multiple non-missing isn't. Maybe add a test where c(1, 2) are given as categories as well (and a comment on the first test that we're testing what happens if there is a missing category would be great!)

Reply:
Great, I added those tests and comments. It looks like I also added a quick check that the subtotal categories match the variable categories.